### PR TITLE
Better admin/config control of mulligans + tweaks

### PR DESCRIPTION
**Pull request by Incoming5643**
Organizes the various continuous_round_x config options into one set of config options: "continuous". Functionality is only there for the same mode that already had it, everything else is always continuous .

Makes another set of config options for which rounds types, if any, use the midround antagonist (mulligan) system. Keep in mind this is by definition a subset of round with continuous set. Again this feature is only present for the roundtypes where I KNOW it works properly, other modes can't use midround antags.

The check antagonist screen will show how the game will behave when all the antagonists die, and it can be changed on the fly by admins, the four possible states are:

End on antagonist death (noncontinuous rounds)
Continue if antagonists die (unsupported continuous by default round)
Continue if antagonists die, creating replacement antagonists (continuous, using midround antags)
Continue if antagonists die, not creating new antagonists (continuous, not using midround antags)

When rounds prepare to mulligan admins will additionally get the option to just keep the round going without any automatically created antags in case they want to run their own shenanigans instead. They can also as before choose to end the round.

Reduces the time between primary antag death and midround antag creation from between 3 to 7 minutes to between 2 to 5 minutes.

Basically the diet coke version of #8586 without all the crazy stuff I haven't tested yet.

### CODE REVIEW

**Comment by MrStonedOne** [comment]
Hey yo @Incoming5643 next time you make something that changes config options. Can you please include details in the pr on how to port over current config options post update for server owners like me?

**Comment by Incoming5643** [comment]
I wasn't expecting the speed merge. But yeah sure here's how it lines up:

New toggles are on when uncommented (no # before the word), so were the old toggles

CONTINUOUS_ROUND_REV = CONTINUOUS REVOLUTION
CONTINUOUS_ROUND_GANG = CONTINUOUS GANG
etc

MIDROUND_ANTAG is a new choice, usable if you wanted a roundtype to be continuous but not by default try to make antags.

The new default settings match up 1:1 with what sybil seemed to be using except that CONTINUOUS_ROUND_WIZ seemed to be off, I think this came about from a bug that I fixed with summon events, so it can be left on if preferred safely.

Just tell me if you outright disable mulliganing for all modes because it's very hard to tell from a player perspective and I can't debug a system that's off.

**Comment by Incoming5643** [comment]
@MrStonedOne also bumping this cause I edited my last message a lot and if you checked in the middle you might have missed something
